pyth-sdk-solana/src/state.rs

```diff
@@ -737,4 +827,59 @@ mod test {
 
         assert_eq!(price_account.get_price_no_older_than(&clock, 1), None);
     }
+
+    #[test]
+    fn test_price_feed_representations_equal() {
+        #[repr(C)]
+        #[derive(Copy, Clone, Debug, Default, PartialEq, Eq)]
+        pub struct OldPriceAccount {
+            pub magic:          u32,
+            pub ver:            u32,
+            pub atype:          u32,
+            pub size:           u32,
+            pub ptype:          crate::state::PriceType,
+            pub expo:           i32,
+            pub num:            u32,
+            pub num_qt:         u32,
+            pub last_slot:      u64,
+            pub valid_slot:     u64,
+            pub ema_price:      Rational,
+            pub ema_conf:       Rational,
+            pub timestamp:      i64,
+            pub min_pub:        u8,
+            pub drv2:           u8,
+            pub drv3:           u16,
+            pub drv4:           u32,
+            pub prod:           Pubkey,
+            pub next:           Pubkey,
+            pub prev_slot:      u64,
+            pub prev_price:     i64,
+            pub prev_conf:      u64,
+            pub prev_timestamp: i64,
+            pub agg:            PriceInfo,
+            pub comp:           [crate::state::PriceComp; 32],
+        }
+
+        let old = OldPriceAccount::default();
+        let new = PriceAccount::default();
+
+        // Equal Sized?
+        assert_eq!(
+            std::mem::size_of::<OldPriceAccount>(),
+            std::mem::size_of::<PriceAccount>(),
+        );
+
+        // Equal Byte Representation?
+        unsafe {
+            let old_b = std::slice::from_raw_parts(
+                &old as *const OldPriceAccount as *const u8,
+                std::mem::size_of::<OldPriceAccount>(),
+            );
+            let new_b = std::slice::from_raw_parts(
+                &new as *const PriceAccount as *const u8,
+                std::mem::size_of::<PriceAccount>(),
+            );
```

Comment:
so 0 equal to 0? doesn't seems like the best test

Reply:
You're right, I've updated the test, albeit in a nasty way (fuzzing would be better) that should verify assigning the fields results in the same byte layout.